src/shader_recompiler/ir/ir_emitter.cpp

```diff
@@ -1359,8 +1359,17 @@ U1 IREmitter::IGreaterThan(const U32& lhs, const U32& rhs, bool is_signed) {
     return Inst<U1>(is_signed ? Opcode::SGreaterThan : Opcode::UGreaterThan, lhs, rhs);
 }
 
-U1 IREmitter::INotEqual(const U32& lhs, const U32& rhs) {
-    return Inst<U1>(Opcode::INotEqual, lhs, rhs);
+U1 IREmitter::INotEqual(const U32U64& lhs, const U32U64& rhs) {
+    if (lhs.Type() != rhs.Type()) {
+        UNREACHABLE_MSG("Mismatching types {} and {}", lhs.Type(), rhs.Type());
+    }
+    switch (lhs.Type()) {
+    case Type::U32:
+    case Type::U64:
+        return Inst<U1>(Opcode::INotEqual, lhs, rhs);
```

Comment:
I believe this actually needs new or changed IR opcode, the opcode `INotEqual` is defined to accept `U32`